setup.py

```diff
@@ -1,24 +1,64 @@
-from __future__ import (absolute_import, division, print_function)
-
 import sys
-import warnings
+from os import path
 
+from setuptools import find_packages, setup
 
-try:
-    from setuptools import setup
-except ImportError:
-    try:
-        from setuptools.core import setup
-    except ImportError:
-        from distutils.core import setup
+import versioneer
 
-from distutils.core import setup
+# NOTE: This file must remain Python 2 compatible for the foreseeable future,
+# to ensure that we error out properly for people with outdated setuptools
+# and/or pip.
+min_version = (3, 6)
+if sys.version_info < min_version:
+    error = """
+hxntools does not support Python {0}.{1}.
+Python {2}.{3} and above is required. Check your Python version like so:
 
-setup(
-    name='hxntools',
-    version="0.4.0",
-    author='Brookhaven National Laboratory',
-    packages=['hxntools', 'hxntools.detectors', 'hxntools.handlers'],
-    install_requires=['numpy', 'h5py', 'databroker', 'nslsii'],
+python3 --version
+
+This may be due to an out-of-date pip. Make sure you have pip >= 9.0.1.
+Upgrade pip like so:
+
+pip install --upgrade pip
+""".format(*(sys.version_info[:2] + min_version))
+    sys.exit(error)
+
+here = path.abspath(path.dirname(__file__))
 
+description = "NSLS-II Hard X-ray Nanoprobe data acquisition tools"
+
+readme_file = path.join(here, 'README.md')
+long_description = description
+if path.isfile(readme_file):
+    with open(readme_file, encoding='utf-8') as f:
+        long_description = f.read()
+
+req_file = path.join(here, 'requirements.txt')
+if path.isfile(req_file):
+    with open(req_file) as f:
+        # Parse requirements.txt, ignoring any commented-out lines.
+        requirements = [line for line in f.read().splitlines()
+                        if not line.startswith('#')]
+
+
+setup(
+    name="hxntools",
+    version=versioneer.get_version(),
+    cmdclass=versioneer.get_cmdclass(),
+    description=description,
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    author="Brookhaven National Laboratory",
+    author_email="DAMA@bnl.gov",
+    url="https://github.com/NSLS-II-HXN/hxntools",
+    python_requires='>={}'.format('.'.join(str(n) for n in min_version)),
+    packages=find_packages(exclude=['docs', 'tests']),
+    include_package_data=True,
+    install_requires=requirements,
+    license="BSD-3-Clause",
+    classifiers=[
+        "Development Status :: 2 - Pre-Alpha",
```

Comment:
Should the development status be a little higher? `Development Status :: 5 - Production/Stable` seems reasonable.

Reply:
Updated via https://github.com/NSLS-II-HXN/hxntools/pull/21/commits/1c1247057b4c2bc34d3500f7b5f4b9fc2debb361.